haskell/nettest/Nettest.hs

```diff
@@ -23,10 +23,10 @@ scTransferScenario
      (Monad m, capsM ~ NettestT m)
   => (OriginationParams -> Storage)
   -> U.Contract
-  -> U.Contract -- Note that we do not distinguish safelist and stablecoin contracts
+  -> U.Contract -- Note that we do not distinguish transferlist and stablecoin contracts
   -> NettestImpl m
   -> m ()
-scTransferScenario constructInitialStorage stablecoinContract safelistContract = uncapsNettest $ do
+scTransferScenario constructInitialStorage stablecoinContract transferlistContract = uncapsNettest $ do
```

Comment:
Is it possible to update this scenario so that instead of transferlist contract it takes `originateTransferlist :: Set (Address, Address) -> Set Address -> m ()`? I. e. a callback that originates transferlist.

The only difficulty here is to construct external transferlist storage value from `Set (Address, Address)` (pairs with allowed transfers) and `Set Address` (permitted receivers). We can make this callback fail when someone tries to originate a transferlist with inconsistent data (e. g. AFAIU it's impossible to permit transfer from Alice to Bob and make `assertReceiver` fail for Bob). In that case we may need to update data we use `initialSafelistStorage`.

Reply:
Btw, I think `scTransferScenario` should be renamed because it tests more than just `transfer` now.